persistence-modules/spring-data-elasticsearch-2/src/test/java/com/baeldung/wildcardsearch/ElasticsearchWildcardServiceIntegrationTest.java

```diff
@@ -0,0 +1,271 @@
+package com.baeldung.wildcardsearch;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch.core.IndexRequest;
+import co.elastic.clients.elasticsearch.indices.CreateIndexRequest;
+import co.elastic.clients.elasticsearch.indices.DeleteIndexRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+@Testcontainers
+class ElasticsearchWildcardServiceIntegrationTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchWildcardServiceIntegrationTest.class);
+
+    @Container
+    static ElasticsearchContainer elasticsearchContainer = new ElasticsearchContainer(
+            "docker.elastic.co/elasticsearch/elasticsearch:8.11.1")
+            .withExposedPorts(9200)
+            .withEnv("discovery.type", "single-node")
+            .withEnv("xpack.security.enabled", "false")
+            .withEnv("xpack.security.http.ssl.enabled", "false");
+
+    @Autowired
+    private ElasticsearchWildcardService wildcardService;
+
+    @Autowired
+    private ElasticsearchClient elasticsearchClient;
+
+    private static final String TEST_INDEX = "test_users";
+
+    @DynamicPropertySource
+    static void configureProperties(DynamicPropertyRegistry registry) {
+        registry.add("elasticsearch.host", elasticsearchContainer::getHost);
+        registry.add("elasticsearch.port", () -> elasticsearchContainer.getMappedPort(9200));
+    }
+
+    @BeforeEach
+    void setUp() throws IOException {
+        // Create test index
+        createTestIndex();
+
+        // Index sample documents
+        indexSampleDocuments();
+
+        // Wait for documents to be indexed
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
```

Comment:
We can't use Thread.sleep - there should be a better way to specifically wait until the documents are indexed - as written this will either be a flaky test or waste CI build time